internal/temporalcommonv1/payload_json.go

```diff
@@ -22,205 +22,587 @@
 
 package common
 
-// "bytes"
-// "encoding/json"
+import (
+	//"bytes"
 
-// jsonpb "go.temporal.io/api/internal/temporaljsonpb"
-// gojsonpb "google.golang.org/protobuf/encoding/protojson"
+	"encoding"
+	"encoding/base64"
+	gojson "encoding/json"
+	"fmt"
+	"reflect"
+	"sort"
+	"strconv"
+	"strings"
+
+	"go.temporal.io/api/internal/protojson/json"
+	"go.temporal.io/api/temporalproto"
+	//	"google.golang.org/protobuf/encoding/protojson"
+)
+
+const (
+	payloadMetadataKey      = "metadata"
+	payloadDataKey          = "data"
+	shorthandMessageTypeKey = "_protoMessageType"
+)
+
+var logfunc func(msg string, args ...any) (int, error) = fmt.Printf
+
+func SetLogfunc(f func(msg string, args ...any) (int, error)) {
+	logfunc = f
+}
+
+var _ temporalproto.JSONPBMaybeMarshaler = (*Payload)(nil)
+var _ temporalproto.JSONPBMaybeMarshaler = (*Payloads)(nil)
+var _ temporalproto.JSONPBMaybeUnmarshaler = (*Payload)(nil)
+var _ temporalproto.JSONPBMaybeUnmarshaler = (*Payloads)(nil)
 
 // !!! This file is copied from internal/temporalcommonv1 to common/v1.
 // !!! DO NOT EDIT at common/v1/payload_json.go.
+func marshalSingular(enc *json.Encoder, value interface{}) error {
```

Comment:
Did you hand-write this marshaling or is it from another library? Why not put this stuff the vendored area somewhere internal with the rest of the vendored code?

Reply:
This is hand-written so there's nothing to vendor. I mean, I can if we want but it's not used elsewhere.

Reply:
Ah, ok, I see now and understand why you wrote this. I do think this is probably too general-purpose to be in a generated file/package, but if you don't want yet another package, ok. I will now review this as newly written code.